### PR TITLE
Adding getting started button

### PR DESCRIPTION
**Pull request by microsoftsam**
Adding getting started button for the widescreen view (in the top right corner). See screenshot:
screen shot 2016-11-11 at 1 33 24 pm

+ Ui layout cleanups
+ Fix for Explorer / Sim buttons showing up before the editor has fully loaded.

The getting started button disappears after the user has loaded any content in the side docs.

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -1665,6 +1665,14 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
         this.shareEditor.show(header);
     }
 
+    gettingStarted() {
+        Util.assert(!this.state.sideDocsLoadUrl);
+        const targetTheme = pxt.appTarget.appTheme;
+        const gettingStartedUrl = targetTheme.gettingStarted || 'getting-started';
```
Comment (pelikhan) [inline webapp/src/app.tsx]: not guaranteed that a getting-started page is available.

File: webapp/src/app.tsx [hunk 2]
```diff
@@ -1754,6 +1762,11 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
                         </div> : undefined }
                     </div>
                 </div>
+                {this.state.sideDocsLoadUrl && targetTheme.gettingStarted ? undefined : 
+                    <div id="getting-started-btn">
+                        <div className="ui bottom attached button green" onClick={() => this.gettingStarted() }>Getting Started</div>
```
Comment (pelikhan) [inline webapp/src/app.tsx]: lf("Getting started")

Comment (pelikhan) [inline webapp/src/app.tsx]: should we use "primary" color?

File: webapp/src/app.tsx [hunk 3]
```diff
@@ -1754,6 +1762,11 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
                         </div> : undefined }
                     </div>
                 </div>
+                {this.state.sideDocsLoadUrl && targetTheme.gettingStarted ? undefined :
```
Comment (pelikhan) [inline webapp/src/app.tsx]: if targetTheme is not defined, this should not be visible.

File: webapp/src/app.tsx [hunk 4]
```diff
@@ -1665,6 +1665,14 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
         this.shareEditor.show(header);
     }
 
+    gettingStarted() {
+        pxt.tickEvent("btn.gettingstarted");
+        const targetTheme = pxt.appTarget.appTheme;
+        Util.assert(!this.state.sideDocsLoadUrl && targetTheme && targetTheme.gettingStarted != undefined);
```
Comment (pelikhan) [inline webapp/src/app.tsx]: or simply

```
Util.assert(!this.state.sideDocsLoadUrl && targetTheme && targetTheme.gettingStarted);
```

File: webapp/src/app.tsx [hunk 5]
```diff
@@ -1754,6 +1762,11 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
                         </div> : undefined }
                     </div>
                 </div>
+                {!this.state.sideDocsLoadUrl && targetTheme && targetTheme.gettingStarted != undefined ?
```
Comment (pelikhan) [inline webapp/src/app.tsx]:
```
{!this.state.sideDocsLoadUrl && targetTheme && targetTheme.gettingStarted
```

File: webapp/src/app.tsx [hunk 6]
```diff
@@ -1754,6 +1762,11 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
                         </div> : undefined }
                     </div>
                 </div>
+                {!this.state.sideDocsLoadUrl && targetTheme && targetTheme.gettingStarted != undefined ? 
+                    <div id="getting-started-btn">
+                        <sui.Button class="bottom attached green" title={lf("Getting Started")} text={lf("Getting Started")} onClick={() => this.gettingStarted() } />
```
Comment (pelikhan) [inline webapp/src/app.tsx]: button text and title should not be the same. Title can be more expressive and use the semantic UI css (see tooltip property)

File: localtypings/pxtarget.d.ts [hunk 7]
```diff
@@ -146,6 +146,7 @@ declare namespace pxt {
         simAnimationEnter?: string;
         simAnimationExit?: string;
         projectGallery?: string;
+        gettingStarted?: string;
```
Comment (pelikhan) [inline localtypings/pxtarget.d.ts]: There is a "sideDocs" property already. Don't we want to reuse or remove that one?

Comment (microsoftsam) [inline localtypings/pxtarget.d.ts]: I thought the sideDocs property enabled/disabled showing content in the side docs. 
My thoughts on the gettingStarted property is to be a flag to enable / disable whether or not the button shows up.

File: webapp/src/app.tsx (hunk 4 above)
Comment (microsoftsam) [inline webapp/src/app.tsx]: Without the != undefined, I get a type error: 
Argument of type 'string' is not assignable to parameter of type 'boolean'

I'd say it has something to do with the assert parameter type

Comment (pelikhan) [inline webapp/src/app.tsx]: Try !! Operator

File: localtypings/pxtarget.d.ts (hunk 7 above)
Comment (pelikhan) [inline localtypings/pxtarget.d.ts]: It was meant exactly for that :) there is hideSideDocs to remove sidedocs.

Comment (pelikhan) [inline localtypings/pxtarget.d.ts]: Need to document schema.

Comment (microsoftsam) [inline localtypings/pxtarget.d.ts]: Agree with documenting schema. I'll start with the ones I know.

Comment (microsoftsam) [inline localtypings/pxtarget.d.ts]: Using sideDoc instead of gettingStarted.

**Pull request merged**